src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -937,7 +756,206 @@ public BigDecimal(String val) {
      * @since  1.5
      */
     public BigDecimal(String val, MathContext mc) {
-        this(val.toCharArray(), 0, val.length(), mc);
+        this((CharSequence) val, mc);
+    }
+
+    private BigDecimal(CharSequence val, MathContext mc) {
```

Comment:
Can we move this constructor to before `parseExp`? This way there will be less diff and it will be easier to compare old and new code side-by-side.